pilot/pkg/keycertbundle/watcher.go

```diff
@@ -62,8 +63,8 @@ func (w *Watcher) RemoveWatcher(id int32) {
 	delete(w.watchers, id)
 }
 
-// SetAndNotify sets the key cert and root cert and notify the watchers.
-func (w *Watcher) SetAndNotify(key, cert, caBundle []byte) {
+// SetAndNotify sets the key, cert, root cert, crl and notifies the watchers.
+func (w *Watcher) SetAndNotify(key, cert, caBundle, crl []byte) {
```

Comment:
It would be much cleaner if we rename this SetAndNotifyCACert and add another function like SetAndNotifyCACRL that just handles CRL - No need to guess based on values being nil I guess

Reply:
Thanks for the review @ramaraochavali. This makes sense. I added new func `SetAndNotifyCACRL` and updated the code accordingly.